src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -508,26 +522,29 @@
             String componentName = attemptLogInformation.getComponentName();
             Set<LogFile> completedFiles = completedLogFilePerComponent.getOrDefault(componentName, new HashSet<>());
 
+
+            // If the component is not configured, delete the uploaded files along with their processing file info.
+            // Otherwise, delete the file based on the component delete configuration.
+            boolean shouldDeleteFile =
+                    !componentLogConfigurations.containsKey(componentName) || componentLogConfigurations.get(
+                            componentName).isDeleteLogFileAfterCloudUpload();
             // Record the last processed file timestamp
             completedFiles.forEach(file -> {
                 updatelastComponentUploadedLogFile(lastComponentUploadedLogFileInstantMap, componentName, file);
+                // This will always remove the processing file information of the uploaded file.
+                this.deleteFileFromGroup(componentName, shouldDeleteFile, file,
+                        attemptLogInformation.getLogFileGroup());
             });
 
-            if (!componentLogConfigurations.containsKey(componentName)) {
-                return;
-            }
-
-            ProcessingFiles processingFiles = processingFilesInformation.get(componentName);
-
             // Delete files based on diskspace
             List<String> deletedHashes =
                     this.diskSpaceManagementService.freeDiskSpace(attemptLogInformation.getLogFileGroup());
 
-            processingFiles.remove(deletedHashes); // Stop tracking files already uploaded
-
-            // Delete after upload
-            ComponentLogConfiguration componentLogConfiguration = componentLogConfigurations.get(componentName);
-            completedFiles.forEach(file -> this.deleteFile(componentLogConfiguration, file));
+            ProcessingFiles processingFiles = processingFilesInformation.get(componentName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Untrusted input is passed to a method that executes arbitrary code, creating a code injection vulnerability. This allows attackers to execute malicious code within the application, potentially leading to data theft or system compromise. Validate input using allowlists with `Arrays.asList().contains()`, use regex matching with `String.matches()`, or parse input safely with `Integer.parseInt()` or `Boolean.parseBoolean()`. [Learn more](https://owasp.org/www-community/attacks/Code_Injection)